spec/services/endpoint/uri_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+RSpec.describe Endpoint::Uri do
+  subject(:use_case) { described_class.new(href, use_case: 'use_case_one', from_date: from_date, to_date: to_date) }
+  let(:href) { '/root/type/sub?matchID=123456' }
+  let(:from_date) { '2021-01-01' }
+  let(:to_date) { '2021-03-31' }
+  it { is_expected.to be_a Endpoint::Uri }
+
+  describe '.for_calling' do
+    subject(:for_calling) { use_case.for_calling }
+
+    context 'when the href just has a matchID' do
+      it { is_expected.to eql '/root/type/sub?matchID=123456' }
+    end
+
+    context 'when the href has a matchID and requires dates' do
+      let(:href) { '/root/type/sub?matchID=123456{&fromDate,toDate}' }
+
+      it { is_expected.to eql '/root/type/sub?matchID=123456&fromDate=2021-01-01&toDate=2021-03-31' }
+    end
+
+    context 'when the href has a matchID and requires tax years' do
+      let(:href) { '/root/type/sub?matchID=123456{&fromTaxYear,toTaxYear}' }
+
+      it { is_expected.to eql '/root/type/sub?matchID=123456&fromTaxYear=2020-21&toTaxYear=2020-21' }
+
+      context 'when the financial years differ' do
```

Comment:
Hi, nothing wrong here, i'm just curious about this work and the whole service.
I'm wondering what will happen if the tax years are different, is that a bad thing? My finance knowledge is weak.

OHHH, is it because of the way Apply takes 3 months of financial data? Thus it could be like March-May which includes 2 different tax years? Is this HMRC service being built for Apply in mind or the whole LAA Digital? What's the purpose of filtering HMRC data by tax years?

Reply:
> What's the purpose of filtering HMRC data by tax years?
It's one of the requirements set by the API, certain endpoints require a fromTaxYear and toTaxYear. 🤷‍♂️